js/wizard.js

```diff
@@ -0,0 +1,437 @@
+  if($(this).is(':checked'))
+  {
+    console.log("wifi");
+    $('#wifi-fields').show();
+  }
+  else {
+    $('#wifi-fields').hide();
+  }
+});
+
+
+$('input[name=wifidhcp]').change(function() {
+  if($(this).is(':checked'))
+  {
+    console.log("wifidhcp");
+    $('#wifi-no-dhcp-fields').hide();
+  }
+  else {
+    $('#wifi-no-dhcp-fields').show();
+  }
+});
+
+$('input[name=wireddhcp]').change(function() {
+  if($(this).is(':checked'))
+  {
+    console.log("wireddhcp");
+    $('#wired-no-dhcp-fields').hide();
+  }
+  else {
+    $('#wired-no-dhcp-fields').show();
+  }
+});
+
+$('input[name=wired]').change(function() {
+  if($(this).is(':checked'))
+  {
+    console.log("wired");
+    $('#wired-fields').show();
+  }
+  else {
+    $('#wired-fields').hide();
+  }
+});
+$('input[name=ntp]').change(function() {
+  if($(this).is(':checked'))
+  {
+    console.log("ntp");
+    $('#ntp-fields').show();
+  }
+  else {
+    $('#ntp-fields').hide();
+  }
+});
+$('input[name=dns]').change(function() {
+  if($(this).is(':checked'))
+  {
+    console.log("dns");
+    $('#dns-fields').show();
+  }
+  else {
+    $('#dns-fields').hide();
+  }
+});
+
+$('#ntp-addbutton').click(function() {
+  var scntDiv = $('#p_scents1');
+  var i = 2;
+
+  if($('#p_scents1 .add-ntp-server') != undefined)
+  i = $('#p_scents1 .add-ntp-server').length + 2;
+  $('<div class="form-group add-ntp-server" id="row_ntp_server_' + i +'"><label for="ntp_server_' + i +'" class="control-label col-xs-3">Server ' + i +'</label><div class="col-xs-6"><input type="text" class="form-control" id="ntp_server_' + i +'" name="ntp_server_' + i +'"></div><span class="remove-button glyphicon glyphicon-minus-sign" id="ntp_remove_server_' + i +'"></span></div></div>').appendTo(scntDiv);
+  $('#ntp_remove_server_' + i).click(function() {
+    $('#row_ntp_server_'+ i).remove();
+    return false;
+  });
+  return false;
+});
+
+$('#dns-addbutton').click(function() {
+  var scntDiv = $('#p_scents2');
+  var i = 2;
+
+  if($('#p_scents2 .add-dns-server') != undefined)
+  i = $('#p_scents2 .add-dns-server').length + 2;
+  $('<div class="form-group add-dns-server" id="row_dns_server_' + i +'"><label for="dns_server_' + i +'" class="control-label col-xs-3">Server ' + i +'</label><div class="col-xs-6"><input type="text" class="form-control" id="dns_server_' + i +'" name="dns_server_' + i +'"></div><span class="remove-button glyphicon glyphicon-minus-sign" id="dns_remove_server_' + i +'"></span></div></div>').appendTo(scntDiv);
+  $('#dns_remove_server_' + i).click(function() {
+    $('#row_dns_server_'+ i).remove();
+    return false;
+  });
+  return false;
+});
+
+
+$('#generateconfig').click(function()
+{
+  $('.form-group').removeClass('has-error');
+  $('.help-block').remove();
+  $('#common_errors').hide();
+
+  // Validate and get values from user
+
+  var validation_errors = [];
+
+  var v1Str = "";
+  var v2Str = "network:\r\n  version: 2\r\n";
+
+  var has_wifi = $('#wifi').is(":checked");
```

Comment:
I think a single object with all vars of the form would be nicer. This could be achieved by serialising the form into JSON. See this [link](https://jsfiddle.net/gabrieleromanato/bynaK/). This way any field can change its `id` or `className` without affecting the script. Obviously the name of each field must maintain intact.